### PR TITLE
[staticcheck] Cleanup deprecations

### PR DESCRIPTION
**Pull request by ajm188**
## Description

- update deprecated grpc dial options
- replace deprecated `strings.Title` with recommended replacement
- RevokedCertificates=>RevokedCertificateEntries
- Stop seeding rand with timestamps, it no longer auto-seeds with a constant (See https://github.com/golang/go/issues/56319)

## Related Issue(s)

#13877 

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on the CI
-   [x] Documentation was added or is not required

### CODE REVIEW

**Comment by dbussink** [comment]
> yeah netutil ones look legit, looking into it

I wonder, if we have distribution issues with `rand.Rand`, would it help to just use `crypto.Rand` instead then to have something with better guarantees?

**Comment by ajm188** [comment]
currently flaky, it seems:

```
➜  vitess git:(andrew/staticcheck-deprecations) ✗ go test -count=10 -run "TestUniformity" ./go/netutil
--- FAIL: TestUniformity (0.00s)
    netutil_test.go:48: actual: 508 diff: 8 e: 500 m: 0.05
    netutil_test.go:48: actual: 350 diff: 16.666666666666686 e: 333.3333333333333 m: 0.1
    netutil_test.go:48: actual: 79 diff: -21 e: 100 m: 0.2
    netutil_test.go:53: missed target weight: expected 100, 79
    netutil_test.go:48: actual: 609 diff: 9 e: 600 m: 0.05
--- FAIL: TestUniformity (0.00s)
    netutil_test.go:48: actual: 529 diff: 29 e: 500 m: 0.05
    netutil_test.go:53: missed target weight: expected 500, 529
    netutil_test.go:48: actual: 332 diff: -1.3333333333333144 e: 333.3333333333333 m: 0.1
    netutil_test.go:48: actual: 99 diff: -1 e: 100 m: 0.2
    netutil_test.go:48: actual: 609 diff: 9 e: 600 m: 0.05
FAIL
FAIL    vitess.io/vitess/go/netutil     0.106s
FAIL
```

**Comment by ajm188** [comment]
gonna try `crypto.Rand`

File: go/netutil/netutil_test.go [hunk 1]
```diff
@@ -54,12 +55,11 @@ func checkDistribution(t *testing.T, data []*net.SRV, margin float64) {
 }
 
 func testUniformity(t *testing.T, size int, margin float64) {
-	rand.Seed(1)
```
Comment (ajm188) [inline go/netutil/netutil_test.go:57]: oh duh, here's my problem

**Review by dbussink — APPROVED** [review]

File: go/netutil/netutil_test.go [hunk 2]
```diff
@@ -24,7 +24,7 @@ import (
 	"testing"
 )
 
-func checkDistribution(t *testing.T, data []*net.SRV, margin float64) {
+func checkDistribution(t *testing.T, rand *rand.Rand, data []*net.SRV, margin float64) {
```
Comment (shlomi-noach) [inline go/netutil/netutil_test.go:27]: Given that the new `golang` behavior is to auto-seed the random generator on startup, why would we want to pass a `rand * Rand` argument to these functions? Wouldn't we just want them to generate random values as opposed to deterministic ones?

Comment (ajm188) [reply go/netutil/netutil_test.go:27]: there were a few test cases that relied on `rand.Seed(1)` before the start of the test, so we update this function to take a rand instance which is either seeded to 1 or to time.Now() (aka "random value")

Comment (shlomi-noach) [reply go/netutil/netutil_test.go:27]: I'm just hesitant to require this additional argument in the function signature. Is it possible to modify the tests to not rely on a reproducible ordering?

Comment (ajm188) [reply go/netutil/netutil_test.go:27]: I hear you, but I am hesitant to change it, and my brief read (without fully digging in/understanding, so admitting that up front) seems to be that it is definitionally dependent on randomness: https://github.com/vitessio/vitess/pull/42/files#diff-b0fbcaba3872685e058eab7c65e0397e27dab0b0df92694e46fbcb67be911f28

Comment (shlomi-noach) [reply go/netutil/netutil_test.go:27]: Yeah, for get it, I notice that these functions are only called internally in the same file (and in the unit test file), so not worth bothering.

**Review by shlomi-noach — APPROVED** [review]

**Pull request merged**